Source/dom/components/Library.js

```diff
@@ -0,0 +1,141 @@
+import { WrappedObject, DefinedPropertiesKey } from '../WrappedObject'
+import { Document } from './Document'
+import { toArray, getURLFromPath } from '../utils'
+import { Types } from '../enums'
+import { Factory } from '../Factory'
+import { wrapObject } from '../wrapNativeObject'
+import { ShareableObject } from './ShareableObject'
+
+const AddStatus = {
+  0: 'ok',
+  1: 'the library has already been added',
+  2: 'the document is not in the new JSON format',
+  3: 'there was a problem reading the asset library file',
+}
+
+/**
+ * A Sketch Library.
+ */
+export class Library extends WrappedObject {
+  constructor(library = {}) {
+    if (!library.sketchObject) {
+      throw new Error('Cannot create a new Library directly')
+    }
+
+    super(library)
+  }
+
+  static getLibraries() {
+    const libraryController = AppController.sharedInstance().librariesController()
+    return toArray(libraryController.libraries()).map(
+      Library.fromNative.bind(Library)
+    )
+  }
+
+  static createLibraryFromDocument(document, path) {
+    if (typeof document === 'string' && !path) {
+      /* eslint-disable no-param-reassign */
+      path = document
+      document = undefined
+      /* eslint-enable */
+    }
+
+    const libUrl = getURLFromPath(path)
+
+    if (document) {
+      const wrappedDocument = wrapObject(document)
+      wrappedDocument.save(path)
+    }
+
+    const libraryController = AppController.sharedInstance().librariesController()
+    const status = libraryController.addAssetLibraryAtURL(libUrl)
+
+    if (status !== 0) {
+      throw new Error(`Error while adding the library: ${AddStatus[status]}.`)
+    }
+
+    // refresh the UI
+    libraryController.notifyLibraryChange(null)
```

Comment:
I don't think this should be necessary. This notification should be sent as a consequence of calling `addAssetLibraryAtURL`